engine/src/main/java/com/datatorrent/stram/plan/logical/LogicalPlanConfiguration.java

```diff
@@ -2277,6 +2279,25 @@ public void prepareDAG(LogicalPlan dag, StreamingApplication app, String name)
     pluginManager.teardown();
   }
 
+  private void prepareDAGAttributes(LogicalPlan dag)
+  {
+    // Consider making all attributes available for DAG construction
+    // EVENTUALLY to be replaced by variable enabled configuration in the demo where the attribute below is used
+    String connectAddress = conf.get(KEY_GATEWAY_CONNECT_ADDRESS);
+    dag.setAttribute(DAGContext.GATEWAY_CONNECT_ADDRESS, connectAddress == null ? conf.get(GATEWAY_LISTEN_ADDRESS) : connectAddress);
+    if (conf.getBoolean(KEY_GATEWAY_USE_SSL, DAGContext.GATEWAY_USE_SSL.defaultValue)) {
```

Comment:
Is it necessary to set other attributes if `connectAddress` is `null`?

Reply:
Should not be needed but if we are trying to stay true to what attributes have been specified then we would set them.

Reply:
What will be a use case? Is there a meaning/usage for username/password/timeout outside of connecting back to gateway?

Reply:
Can't think of one. Are there any side effects to not including them when they are specified in the config file?

Reply:
Probably not, but they look to me as the dependent attributes that do not have meaning if the gateway URL is not available.

Reply:
We don't do any attribute dependency handling anywhere afaik. So this would be a first if we do this. It might be better to do this in a more systematic way such as ability to describe these relationships in the attribute specification itself as opposed to hard-coding this in the code.